src/main/java/net/william278/velocitab/packet/TeamsPacketAdapter.java

```diff
@@ -24,24 +24,48 @@
 import io.netty.buffer.ByteBuf;
 import lombok.Getter;
 import lombok.RequiredArgsConstructor;
+import net.kyori.adventure.text.Component;
 import net.william278.velocitab.Velocitab;
-import org.apache.commons.text.StringEscapeUtils;
 import org.jetbrains.annotations.NotNull;
 
+import java.util.Map;
+import java.util.Optional;
 import java.util.Set;
+import java.util.concurrent.ConcurrentHashMap;
 
 @Getter
 @RequiredArgsConstructor
 public abstract class TeamsPacketAdapter {
 
     private final Velocitab plugin;
     private final Set<ProtocolVersion> protocolVersions;
+    private final Map<Component, ChatCache> chatCache = new ConcurrentHashMap<>();
```

Comment:
@alexdev03 Instead of managing this cache ourself, why don't we use an `ExpiringMap`?